src/components/structures/RoomView.js

```diff
@@ -93,7 +97,7 @@ module.exports = React.createClass({
     },
 
     getInitialState: function() {
-        var room = this.props.roomId ? MatrixClientPeg.get().getRoom(this.props.roomId) : null;
+        var room = MatrixClientPeg.get().getRoom(this.props.roomAlias);
```

Comment:
Indeed. If it's an alias, we'll show the preview with a 'join' link - as per the comments on propTypes.